### PR TITLE
Automate unit testing with Docker container

### PR DESCRIPTION
**Pull request by nmldiegues**
This configures Travis to run unit tests with support of a Docker container running an image set up with R, Rserve, Caret, Java and Maven (which has been pushed
to DockerHub: https://hub.docker.com/r/nmldiegues/rserve-caret-java/).

### CODE REVIEW

**Comment by nmldiegues** [comment]
@henriquevcosta  How come the unit tests did not fail in the automated checks? (we haven't set up anything yet to run the couple of commands indicated in the commit message, so I'd expect some check to fail as the UTs will fail to connect to Rserve).

**Comment by henriquevcosta** [comment]
I will now submit a version with failing tests to make sure this fails.

File: .travis.yml [hunk 1]
```diff
@@ -9,8 +14,16 @@ branches:
     - /^\d+\.\d+(\.\d+)?(-\S*)?$/
     #hotfix branches (not being too strict on purpose)
     - /^hf-\.*$/
+    
+before_install:
+  - docker pull nmldiegues/rserve-caret-java:latest
+  - docker run -v $(pwd):/opt/openml -d --name test-container nmldiegues/rserve-caret-java:latest
+
+install:
+  - docker exec -it test-container mvn install -f /opt/openml/pom.xml -DskipTests=true -Dmaven.javadoc.skip=true -B -V
```
Comment (nmldiegues) [inline .travis.yml]: Why do we need this besides the "script" below?

File: pom.xml [hunk 2]
```diff
@@ -251,16 +251,23 @@
...
-                <artifactId>cobertura-maven-plugin</artifactId>
-                <version>2.7</version>
-                <configuration>
-                    <formats>
-                        <format>html</format>
-                        <format>xml</format>
-                    </formats>
-                    <check/>
-                </configuration>
+                <groupId>org.jacoco</groupId>
```
Comment (nmldiegues) [inline pom.xml]: What's motivating the change? Is it related with the new way we want to run tests?

Comment (henriquevcosta) [reply pom.xml]: Just discovered that cobertura maven plugin doesn't work properly with Java 8 (it's a bug that's been fixed for over 6 months but no release has been made). Most classes weren't being properly instrumented. Will replicate on the other repo.

File: .travis.yml (hunk 1 above)
Comment (henriquevcosta) [reply .travis.yml]: I though that way at least we only download maven dependencies once 😛 (will download again for deployments, but those are less frequent).

**Comment by nmldiegues** [comment]
I am still the author of the PR so I cannot approve it. But I definitely want to; cool stuff!

**Pull request merged**